src/python/pants/backend/tools/semgrep/rules_integration_test.py

```diff
@@ -0,0 +1,424 @@
+    result = results[0]
+    assert "find-bad-pattern" in result.stdout
+    assert "Ran 1 rule on 1 file: 1 finding" in result.stderr
+    assert result.exit_code == SEMGREP_ERROR_FAILURE_RETURN_CODE
+    assert result.report == EMPTY_DIGEST
+
+    # ...but there's a pants warning about it...
+    assert "Semgrep does not obey .semgrepignore outside the working directory" in caplog.text
+    assert f"{DIR}/.semgrepignore" in caplog.text
+
+    caplog.clear()
+
+    # ... that can be silenced
+    results = run_semgrep(
+        rule_runner,
+        [tgt],
+        extra_args=["--semgrep-acknowledge-nested-semgrepignore-files-are-not-used"],
+    )
+    assert not caplog.records
+
+
+def test_partition_by_config(rule_runner: RuleRunner) -> None:
+    file_dirs = []
+
+    def file___(dir: str) -> dict[str, str]:
+        file_dirs.append(dir)
+        return {
+            f"{dir}/file.txt": GOOD_FILE,
+            f"{dir}/BUILD": """file(name="f", source="file.txt")""",
+        }
+
+    def semgrep(dir: str) -> dict[str, str]:
+        return {f"{dir}/.semgrep.yml": RULES, f"{dir}/BUILD": """semgrep_rule_sources(name="s")"""}
+
+    rule_runner.write_files(
+        {
+            # 'y'/'n' indicates whether that level has semgrep config
+            **semgrep("y"),
+            **file___("y/dir1"),
+            **file___("y/dir2"),
+            **file___("y/n/dir1"),
+            **file___("y/n/dir2"),
+            **semgrep("y/y"),
+            **file___("y/y/dir1"),
+            **file___("y/y/dir2"),
+            **file___("n"),
+        }
+    )
+
+    field_sets = tuple(
+        SemgrepFieldSet.create(rule_runner.get_target(Address(dir, target_name="f")))
+        for dir in file_dirs
+    )
+
+    partitions = rule_runner.request(
+        Partitions[SemgrepFieldSet, PartitionMetadata],
+        [SemgrepRequest.PartitionRequest(field_sets)],
+    )
+
+    sorted_partitions = sorted(
+        (
+            sorted(field_set.address.spec for field_set in partition.elements),
+            sorted(f.address.filename for f in partition.metadata.config_files),
+        )
+        for partition in partitions
+    )
+
+    assert sorted_partitions == [
+        (
+            ["y/dir1:f", "y/dir2:f", "y/n/dir1:f", "y/n/dir2:f"],
+            ["y/.semgrep.yml"],
+        ),
+        (
+            ["y/y/dir1:f", "y/y/dir2:f"],
+            ["y/.semgrep.yml", "y/y/.semgrep.yml"],
+        ),
+        # n: doesn't appear in any partition
+    ]
+
+
+def test_skip(rule_runner: RuleRunner) -> None:
+    rule_runner.write_files(BAD_FILE_LAYOUT)
+    tgt = rule_runner.get_target(Address(DIR, target_name="f"))
+
+    results = run_semgrep(rule_runner, [tgt], extra_args=["--semgrep-skip"])
+    assert not results
+
+
+@pytest.mark.xfail(
+    reason=""" TODO: --semgrep-force does rerun the underlying process, but the LintResult's
+    contents are the same (same stdout etc.), these are deduped, and thus we cannot detect the
+    rerun"""
```

Comment:
This is the same issue as:

https://github.com/pantsbuild/pants/blob/71a55b44b7941a2e78e3e4133194304cb558a438/src/python/pants/backend/shell/shunit2_test_runner_integration_test.py#L194-L196

I suspect the equivalent test works for pytest because its output is non-deterministic.